src/linear_samplers/sampler_helpers/hadamard.jl

```diff
@@ -0,0 +1,46 @@
+"""
+fwht!(x::AbstractVector; signs=ones(Bool, size(x)), scaling = 1)  
+
+Performs an in-place Fast Walsh Hadamard Transform (FWHT). `signs` allows the user to input a booleanvector that can flip the signs in the transform. `scaling` allows the user to scale the result of the transform. 
+
+!!! Note: To avoid log computation at every call the function does not check that the dimension is a power of 2. This must be done by a separate function at an earlier point.
```

Comment:
What do you mean by `log computation`? I think you should just leave this as `To reduce costs, this function avoids checking that the dimension is a power of 2`